src/property-filter/__tests__/property-filter.test.tsx

```diff
@@ -303,12 +308,54 @@ describe('property filter parts', () => {
         expect.objectContaining({ filteringText: '123', filteringProperty: stateProperty, filteringOperator: '=' }),
       ]);
 
+      createWrapper(container).find('#focus-target')!.focus();
+
       wrapper.findClearButton()!.click();
       jest.advanceTimersByTime(1000);
+
       expect(loadItemCalls).toEqual([
         expect.objectContaining({ filteringText: '' }),
         expect.objectContaining({ filteringText: '' }),
         expect.objectContaining({ filteringText: '123', filteringProperty: stateProperty, filteringOperator: '=' }),
+        expect.objectContaining({ filteringText: '123', filteringProperty: stateProperty, filteringOperator: '=' }),
+        expect.objectContaining({ filteringText: '' }),
+        expect.objectContaining({ filteringText: '' }),
+      ]);
+    });
+
+    test('calls onLoadItems on clear when filter by text and asyncProperties=true', () => {
+      const loadItemCalls: PropertyFilterProps.LoadItemsDetail[] = [];
+      const { propertyFilterWrapper: wrapper, container } = renderComponent({
+        onLoadItems: ({ detail }) => loadItemCalls.push(detail),
+        filteringStatusType: 'pending',
+        asyncProperties: true,
+      });
+
+      wrapper.focus();
+      expect(loadItemCalls).toEqual([
+        expect.objectContaining({ filteringText: '' }),
+        expect.objectContaining({ filteringText: '' }),
+      ]);
+
+      wrapper.setInputValue('stat');
+      expect(loadItemCalls).toHaveLength(2);
+
+      jest.advanceTimersByTime(1000);
+      expect(loadItemCalls).toEqual([
+        expect.objectContaining({ filteringText: '' }),
+        expect.objectContaining({ filteringText: '' }),
+        expect.objectContaining({ filteringText: 'stat' }),
+      ]);
+
+      createWrapper(container).find('#focus-target')!.focus();
+
+      wrapper.findClearButton()!.click();
+
```

Comment:
If you add `jest.advanceTimersByTime(1000);` - the expected load-items event is sent w/o any changes to the component's code.

Reply:
true, added and updated the test. The expected event was there because `filteringStatusType` was still pending. That triggered `fireLoadMoreOnScroll`. But it should have been `finished`, as there are no more results to load

Reply:
Should we then update the "calls onLoadItems on clear when filter by property and asyncProperties=true" test accordingly? Rn it looks like the behaviour is different between parsed and free-text input.